frontend/src/packages/editor/components/EditorBlock.tsx

```diff
@@ -46,6 +48,51 @@ const Text = styled.span<{
   background-color: ${(props) => props.code ? "#eee" : "#fff"};
 `;
 
+
+const CheckListItemElement = ({children, attributes, leaf}: any) => {
+  const readOnly = useReadOnly()
+  const [checked, setChecked] = React.useState(false)
+  return (
+    <div {...attributes} style={{display: 'flex', alignItems: 'left'}}>
+      <div
+        style={{ userSelect: "none" }}
+        contentEditable={false}
+      >
+        <span
+          contentEditable={false}
+          style={{
+            marginRight: '0.75em'
+          }}
+        >
+          <input
+            type="checkbox"
+            checked={checked}
+            onChange={event => {
+              // Figure out why ReactEditor.findPath does not work
+              // const path = ReactEditor.findPath(editor as ReactEditor, leaf);
```

Comment:
Your current implementation works so you can just remove these comments but if you wanted to know why the ReactEditor.findPath() doesn't work, i think these could possibly be the reasons:
1. the editor needs to be passed in from the EditorBlock component
2. this findPath function is specifically finding a **node** (https://docs.slatejs.org/libraries/slate-react/react-editor#reacteditor.findpath-editor-reacteditor-node-node-path) but here you are rendering the checklists as leaves. This means that the function will never return the current **leaf**. You can resolve this by changing the checklists into nodes. NOTE: you can think of a node as sort of a big wrapper/container whilst leaves are sort of the children with decorations. However, I do have some doubts about this cause in that case it should in theory find the leaf level text node surrounding this lead? Not completely sure without delving into it too deeply.
3. This is probably the most likely reason: the structure of our editor/leaves/nodes does not follow typical slate operations and therefore any functions that involve traversing through the editor path is doomed to fail